### PR TITLE
CommonConnection for async and non-async 

### PR DESCRIPTION
**Pull request by rahulKQL**
### Description
This change contains addition of CommonConnection, which is implemented in AbstarctBigtableConnection & BigtableAsyncConnection.

### Imapact of This PR
After this PR, BigtableAdmin & BigtableAsyncAdmin would have same argument in the constructor, so that a same connection instance can be used to create either of the Admin classes.

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 1]
```diff
@@ -700,10 +701,12 @@ public Configuration getConfiguration() {
   /** {@inheritDoc} */
   @Override
   public List<HRegionInfo> getTableRegions(TableName tableName) throws IOException {
-    List<HRegionInfo> regionInfos = new ArrayList<>();
-    for (HRegionLocation location : connection.getRegionLocator(tableName).getAllRegionLocations()) {
+    List<HRegionInfo> regionInfos = Collections.emptyList();
+    BigtableRegionLocator regionLocator = new BigtableRegionLocator(tableName, options, connection.getSession().getDataClient());
```
Comment (sduskis) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: Can we please add a `CommonConnection.getAllRegionInfos(TableName)` method instead?

**Comment by sduskis** [comment]
@rahulKQL, a bunch of HBase 2 tests fail.  We need to fix those.

**Comment by rahulKQL** [comment]
Hi @sduskis,
I ran all 4 integration-test profile, It's all building successfully in my system. Could you please re-run the **Hbase-2**  test on [code_duplication_removal](https://github.com/rahulKQL/cloud-bigtable-client/tree/code_duplication_removal) branch.

**Comment by rahulKQL** [comment]
This commit now contains: 

- Changes reverted  related to the `getSampledRowKeysAdapter()`, as it is still having **compatibility issue**.
- Also added `getAllRegionsInfo()`, on `BigtableAdmin` & `BigtableAsyncAdmin`. 
- Ran test cases against actual(_node-client-performance_) HBase 1.x & 2.x both **passed**. 

@sduskis Please have a look and let me know if anything is required.

**Pull request merged**